Feature/gazebo lidar plugins

zarmomin opened the pull request:
Adds parametrized and extendable URDF model of a Velodyne HDL-32 Lidar. Further implements a control plugin to spin the Lidar at constant speed as well as a plugin to publish the beams in XYZ coordinates while maintaining angular information.
Adds a launch file to demonstrate it: `roslaunch rotors_description xacrodisplay_velodyne.launch`
Optionally show the assembled pointcloud: `roslaunch rotors_description xacrodisplay_velodyne.launch assembler_enabled:=True`

ZacharyTaylor committed:
fix not including message types

zarmomin committed:
adds model urdf files for velodyne hdl32

zarmomin committed:
update gitignore

zarmomin committed:
Adds Plugins to control continous joint and to publish lidar data

zarmomin committed:
reverts changes made in pending PR #522 to be independent of it

zarmomin committed:
Makes use of velodyne assembler optional

fmina reviewed (changes requested):
Minor changes. 
It doesn't pass all the checks to merge. Try to merge locally first to find conflicts.

zarmomin commented:
Regarding the checks: It appears that Jenkins fails on 14.04 since it cannot locate libgeographic-dev, and the last successful build was > 2 months ago.

zarmomin commented on rotors_description/rviz/hdl32.rviz:
```diff
@@ -0,0 +1,196 @@
+Panels:
```
So into rotors_description/resources/rviz/...?

zarmomin commented on rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:
```diff
@@ -0,0 +1,85 @@
+    ROS_INFO("No joint name provided, therefore the first joint is used.");
+  }
+  // Setup a P-controller, with a gain of 0.1.
+  this->pid_ = common::PID(0.1, 0, 0);
```
I'll make all of them adjustable, thanks for pointing that out!

zarmomin committed:
Adresses PR comments. Removes redundant functionality

zarmomin committed:
Bugfix: actually attach lidar sensor to parent frame

zarmomin committed:
Adds a VLP-16 lidar model

zarmomin committed:
updates the motor controller to be based off a Dynamixel implementation

zarmomin committed:
adds robot namespace to velodyne xacros

zarmomin committed:
Merge branch 'vlp-16' into feature/gazebo_lidar_plugins

zarmomin committed:
adds hovering example launch file with lidar mounted

zarmomin committed:
adds the os1 model

zarmomin committed:
updates hovering launchfile

zarmomin committed:
fixes joint-publisher frequency to make simulation work more reliably.

zarmomin committed:
updates unitbox world to properly work with rotors

zarmomin committed:
fixes namespaces in os-1 models and related plugins

zarmomin committed:
fixes vlp-16 namespacing

zarmomin committed:
fixes HDL-32 namespaces and demo files

zarmomin committed:
Merge local branch into feature/gazebo_lidar_plugins

zarmomin commented:
@fmina @inkyusa I added a couple of things, it'd be great if you could take another look.

fmina commented on rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:
```diff
@@ -0,0 +1,165 @@
+  std::string joint_state_topic_name;
+  getSdfParam<std::string>(sdf, "joint_state_topic_name",
+                           joint_state_topic_name, "joint_state");
+  joint_->SetPosition(0, current_motor_state_.current_pos_rad);
```
SetPosition, GetParam, SetParam seem to be incompatible with gazebo 2, that's why Jenkins fail on ubuntu trusty.
`trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp: In member function 'virtual void gazebo::ContinousJointControllerPlugin::Load(gazebo::physics::ModelPtr, sdf::ElementPtr)':
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:92:11: error: 'class gazebo::physics::Joint' has no member named 'SetPosition'
   joint_->SetPosition(0, current_motor_state_.current_pos_rad);
           ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp: In member function 'void gazebo::ContinousJointControllerPlugin::UpdateMotor(const MotorState&)':
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:139:15: error: 'class gazebo::physics::Joint' has no member named 'GetParam'
   if (joint_->GetParam("vel", 0) != current_motor_state_.velocity_rad_s) {
               ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:140:13: error: 'class gazebo::physics::Joint' has no member named 'SetParam'
     joint_->SetParam("vel", 0, current_motor_state_.velocity_rad_s);
             ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:145:13: error: 'class gazebo::physics::Joint' has no member named 'SetParam'
     joint_->SetParam("fmax", 0, read_motor_state.torque_limit);
             ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:147:13: error: 'class gazebo::physics::Joint' has no member named 'SetParam'
     joint_->SetParam("fmax", 0, 0.0);`

zarmomin replied:
Thanks, it got farther than it did last time. I updated the CMakeLists to skip this plugin if the gazebo version is too low

fmina commented:
minor comment on Jenkins failure, otherwise LGTM

zarmomin committed:
Triggers Continous Joint Controller plugin only if gazebo version > 4

zarmomin committed:
moves the rviz config files into a 'resources' folder

zarmomin committed:
adds a different implementation of an os1 plugin

makes it capable of hitting the high spinning frequencies without crashing gazebo. tradeoff is a lack of information between the scans

ffurrer commented on rotors_description/urdf/velodyne_VLP_16.urdf.xacro:
```diff
@@ -0,0 +1,163 @@
+        <inertia ixx="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" ixy="0" ixz="0"
+                 iyy="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" iyz="0"
+                 izz="${0.5 * m1 * (r1*r1)}"/>
+      </inertial>
```
yup same as above

ffurrer commented on rotors_description/urdf/velodyne_VLP_16.urdf.xacro:
```diff
@@ -0,0 +1,163 @@
+        <inertia ixx="${(m2 * (3.0*r2*r2 + l2*l2)) / 12.0}" ixy="0" ixz="0"
+                 iyy="${(m2 * (3.0*r2*r2 + l2*l2)) / 12.0}" iyz="0"
+                 izz="${0.5 * m2 * (r2*r2)}"/>
+      </inertial>
```
dito

ffurrer commented on rotors_description/urdf/ouster_os1.urdf.xacro:
```diff
@@ -0,0 +1,162 @@
+        <inertia ixx="${(m2 * (3.0*r2*r2 + l2*l2)) / 12.0}" ixy="0" ixz="0"
+                 iyy="${(m2 * (3.0*r2*r2 + l2*l2)) / 12.0}" iyz="0"
+                 izz="${0.5 * m2 * (r2*r2)}"/>
+      </inertial>
```
same

ffurrer commented on rotors_description/urdf/ouster_os1.urdf.xacro:
```diff
@@ -0,0 +1,162 @@
+        <inertia ixx="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" ixy="0" ixz="0"
+                 iyy="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" iyz="0"
+                 izz="${0.5 * m1 * (r1*r1)}"/>
+      </inertial>
```
same

ffurrer commented on rotors_description/urdf/ouster_os1.urdf.xacro:
```diff
@@ -0,0 +1,162 @@
+        <inertia ixx="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" ixy="0" ixz="0"
+                 iyy="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" iyz="0"
+                 izz="${0.5 * m1 * (r1*r1)}"/>
+      </inertial>
```
Maybe create something like https://github.com/ethz-asl/rotors_simulator/blob/master/rotors_description/urdf/multirotor_base.xacro#L31

ffurrer commented on rotors_description/urdf/velodyne_VLP_16.urdf.xacro:
```diff
@@ -0,0 +1,163 @@
+        <inertia ixx="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" ixy="0" ixz="0"
+                 iyy="${(m1 * (3.0*r1*r1 + l1*l1)) / 12.0}" iyz="0"
+                 izz="${0.5 * m1 * (r1*r1)}"/>
+      </inertial>
```
same

ffurrer reviewed (approved):
lgtm, please run clang-format before merging

zarmomin committed:
further parametrization of the lidar models

zarmomin committed:
clang-format

zarmomin committed:
adds descriptions

zarmomin committed:
fix

zarmomin committed:
Merge branch 'feature/gazebo_lidar_plugins' into os1_plugin

zarmomin committed:
Makes the fixed OS-1 the default. Moves the spinning one into a different file

zarmomin committed:
clang-format

zarmomin committed:
fixes the os-1

zarmomin committed:
fixes the spinning os1. updates xacro ros-link declarations

zarmomin committed:
Merge branch 'master' into os1_plugin

zarmomin committed:
Merge remote-tracking branch 'origin/feature/nscheidt_os1plugin' into feature/gazebo_lidar_plugins

fmina reviewed (approved):
LGTM

michaelpantic commented:
We are changing our codebase to gazebo9 soon. Please wait with merging this. 
@zarmomin  can you address the comments of @ffurrer until mid next week? Then I would rebase this to gazebo9 branch and test it.

zarmomin commented:
@michaelpantic I addressed Fadris comments in https://github.com/ethz-asl/rotors_simulator/pull/527/commits/bf67a530b4c31aaad0c44c8b1e758dcdef319887 and https://github.com/ethz-asl/rotors_simulator/pull/527/commits/44910db2c58121b62579e0109e0f6ad18a0b2756
You should be good to go with gazebo 9 :-)

zarmomin committed:
makes number of beams adjustable for the os1

inkyusa committed:
fixing typo.

marselomeri commented:
having problems building this~
**Conflicting files** 
gazebo_os1_plugin.h
gazebo_os1_plugin.cpp
![Screenshot from 2019-04-29 19-05-35](https://user-images.githubusercontent.com/41054948/56913398-dcf04400-6ab1-11e9-98fc-bd9a6470f4d7.png)

zarmomin commented:
It looks like you have quite a lot of files from other branches in there (all the .bak extensions etc). If you check out this PR/branch and remove all those remaining files in git it should build just fine.

michaelpantic commented:
@zarmomin Can you merge in current master into this branch? Then we can check this again and merge.
